controller/src/main/java/org/jboss/as/controller/AttributeMarshallers.java

```diff
@@ -55,12 +55,18 @@ public boolean isMarshallable(AttributeDefinition attribute, ModelNode resourceM
 
         @Override
         public void marshallAsElement(AttributeDefinition attribute, ModelNode resourceModel, boolean marshallDefault, XMLStreamWriter writer) throws XMLStreamException {
-            if (!resourceModel.hasDefined(attribute.getName())) {
+            if (!resourceModel.has(attribute.getName())) {
```

Comment:
The initial if (!resourceModel.has(attribute.getName())) {
was there as part of the not using setEmptyObject() and is at this point completely irrelevant and I have removed it.